httpx/concurrency/trio.py

```diff
@@ -0,0 +1,214 @@
+import functools
+import math
+import ssl
+import typing
+from types import TracebackType
+
+import trio
+
+from ..config import PoolLimits, TimeoutConfig
+from ..exceptions import ConnectTimeout, PoolTimeout, ReadTimeout, WriteTimeout
+from .base import (
+    BaseBackgroundManager,
+    BaseEvent,
+    BasePoolSemaphore,
+    BaseQueue,
+    BaseStream,
+    ConcurrencyBackend,
+    TimeoutFlag,
+)
+
+
+def _or_inf(value: typing.Optional[float]) -> float:
+    return value if value is not None else float("inf")
+
+
+class Stream(BaseStream):
+    def __init__(self, stream: trio.abc.Stream, timeout: TimeoutConfig) -> None:
+        self.stream = stream
+        self.timeout = timeout
+        self.is_eof = False
+        self.write_buffer = b""
+
+    def get_http_version(self) -> str:
+        if not isinstance(self.stream, trio.SSLStream):
+            return "HTTP/1.1"
+
+        ident = self.stream.selected_alpn_protocol()
+        if ident is None:
+            ident = self.stream.selected_npn_protocol()
```

Comment:
@florimondmanca I'm not sure if NPN is worth considering any more, I've removed the NPN parts from Hypercorn.

Reply:
Not worth it as in — NPN isn't in use anymore/is obsolete? I think @tomchristie originally implemented this check in the asyncio backend.

Reply:
Yep, from https://gitlab.com/pgjones/hypercorn/merge_requests/16, which I believe to be correct, 
> It's my understanding that NPN was for SPDY, ALPN is used by non-ancient http/2 clients. Very old http/2 clients might speak http/1 to hypercorn after this change.

Reply:
Cool. I think this can be discussed in a separate issue, as it already affects the existing `AsyncioBackend`. :-)

Reply:
https://github.com/encode/httpx/issues/280